src/test/java/org/haffson/adventofcode/days/day01/Day01Test.java

```diff
@@ -4,34 +4,43 @@
 import org.junit.Assert;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.test.context.junit4.SpringRunner;
 
+
 @RunWith(SpringRunner.class)
+@SpringBootTest
 public class Day01Test {
 
-    @MockBean
+    @Autowired
     private FileReaders fileReaders;
 
-    @Test
+    @Value("${day1.file}")
+    private String filePath;
+
     public void testGetDay() {
```

Comment:
See the comment above about testing the business logic separately 😄 
In addition, you could add tests that verify these simple example inputs from the advent of code problems. The problems will get more complicated and also will compute for longer than a few seconds. If you have these (and many) small tests with small datasets, you can verify that your code works in principle, before running it with the large input files.

Reply:
You're right. I need to write more tests for the special cases.